src/runtime/storageNG/storage-key-factory.ts

```diff
@@ -24,45 +27,62 @@ export type StorageKeyCreator = (options: StorageKeyOptions) => StorageKey;
  * when laying arcs into a shared storage.
  */
 export class StorageKeyFactory {
```

Comment:
Does the factory need to be separate from the CapabilitiesResolver, now that we're using findStorageKeyProtocols(Capabilities) to essentially do all the work of the resolver?

Reply:
The factory is a simple registry, and the CapabilitiesResolver contains (or will contain) the logic of choosing the right StorageKeyCreator. Also as we add Capabilities, there could be additional functionality added to the resolver. So I'd probably prefer to keep them separate, until we know more. 
Will merge as is for now, but happy to rework in a follow up PR, if you prefer.

Reply:
But the call to resolve capabilities goes through the factory, which contradicts "is a simple registry".